mdetsims/sim_utils.py

```diff
@@ -616,13 +646,33 @@ def _get_band_objects(self):
             elif self.gal_type == 'ground_galsim_parametric':
                 gals = self._get_gal_ground_galsim_parametric()
             elif self.gal_type == 'wldeblend':
-                gals = self._get_gal_wldeblend()
+                gals, rind = self._get_gal_wldeblend()
+
+                # add shear info to calculate the final position
+                if 0. < self.OneSq_cat[rind] < 0.2:
+                    z_population[i] = 0
+                elif 0.2 < self.OneSq_cat[rind] < 0.4:
+                    z_population[i] = 1
+                elif 0.4 < self.OneSq_cat[rind] < 0.6:
+                    z_population[i] = 2
+                elif 0.6 < self.OneSq_cat[rind] < 0.8:
+                    z_population[i] = 3
+                elif 0.8 < self.OneSq_cat[rind] < 1.0:
+                    z_population[i] = 4
+                else:
+                    z_population[i] = -1
             else:
                 raise ValueError('gal_type "%s" not valid!' % self.gal_type)
 
             # compute the final image position
             if self.shear_scene:
-                sdx, sdy = np.dot(self.shear_mat, np.array([dx, dy]))
+                if self.flag_ex:
+                    if z_population[i] == 1:
```

Comment:
This line is wrong. You are always applying the other shear to the z_pop == 1 interval.